### ISSUE TITLE
HashLocationStrategy has wrong order for providers

### ISSUE THREAD

**Issue by abierbaum** [angular/angular.io#626]
When I followed the documentation here: https://angular.io/docs/ts/latest/guide/router.html#!#browser-url-styles  for setting up a HashLocationStrategy it didn't work.

If I switch the bootstrap provider list order around to put the HashLocationStrategy second in the list, then everything seems to work.

```
bootstrap(AppComponent, [
  ROUTER_PROVIDERS,
  provide(LocationStrategy,
         {useClass: HashLocationStrategy}) // ~/src/#/crisis-center/
]);
```

**Comment by wardbell** [comment]
It shouldn't matter but I'll update the example.

I refiled as [an Angular issue](https://github.com/angular/angular/issues/6201)

**Comment by wardbell** [comment]
Actually it DOES matter.  The `ROUTE_PROVIDERS` sets the default `LocationStrategy` to the `PathLocationStrategy`. We must provide the overriding alternative `HashLocationStrategy` later in the bootstrap providers array for it to take effect.

Pascal caught this in his comment to the [Angular issue](https://github.com/angular/angular/issues/6201).  I've updated the doc.

**Comment by wardbell** [comment]
Closed by pr #637
